pkg/secretlock/secretsplitlock/secret_split_lock.go

```diff
@@ -59,15 +68,37 @@ func New(secret []byte, params *HubAuthParams, options ...Option) (secretlock.Se
 		options[i](opts)
 	}
 
+	doOnce.Do(func() {
+		opts.Logger.Infof("Initializing cache...")
+		cache = gcache.New(0).
+			LoaderFunc(func(key interface{}) (interface{}, error) {
+				uriAndToken := strings.Split(fmt.Sprintf("%s", key), "&")
+
+				return fetchSecretShare(uriAndToken[0], uriAndToken[1], opts.HTTPClient, opts.Logger)
+			}).
+			Expiration(time.Minute).
+			Build()
+
+		opts.Logger.Infof("Initializing cache done.")
+	})
+
 	if secret == nil {
 		return nil, errors.New("empty secret share")
 	}
 
-	otherSecret, err := fetchSecretShare(params.URL, params.APIToken, params.Subject, opts.HTTPClient, opts.Logger)
+	uri := fmt.Sprintf("%s%s?sub=%s", params.URL, hubAuthSecretPath, url.QueryEscape(params.Subject))
+	token := fmt.Sprintf("Bearer %s", base64.StdEncoding.EncodeToString([]byte(params.APIToken)))
+
+	cachedSecret, err := cache.Get(uri + "&" + token)
```

Comment:
Using `&` as a delimiter is a bad idea when one of the components is a URI.

It's safer if you first `url.PathEscape(hubAuthSecretPath)` and then use a [non-reserved](https://tools.ietf.org/html/rfc3986#section-2.2) character as delimiter (eg. `|`).

Reply:
Using a struct as a key also works:

type cacheKey struct {
    uri   string
    token string
}